datajoint/external.py

```diff
@@ -6,6 +6,203 @@
 from .declare import STORE_HASH_LENGTH, HASH_DATA_TYPE
 
 
+class ExternalFileHandler:
+
+    _handlers = {}  # handler mapping - statically initialized below.
+
+    def __init__(self, store, database):
+        self._store = store
+        self._database = database
+        self._spec = config[store]
+
+    @classmethod
+    def get_handler(cls, store, database):
+
+        if store not in config:
+            raise DataJointError(
+                'Store "{store}" is not configured'.format(store=store))
+
+        spec = config[store]
+
+        if 'protocol' not in spec:
+            raise DataJointError(
+                'Store "{store}" config is missing the protocol field'
+                .format(store=store))
+
+        protocol = spec['protocol']
+
+        if protocol not in cls._handlers:
+            raise DataJointError(
+                'Unknown external storage protocol "{protocol}" for "{store}"'
+                .format(store=store, protocol=protocol))
+
+        return cls._handlers[protocol](store, database)
+
+    def check_required(self, store, storetype, required):
```

Comment:
`storetype` does not seem to be used for anything.  If you are using specialized classes, then the class itself represents the store type.